cmake/generic_toolchain.cmake

```diff
@@ -56,30 +46,16 @@ assert(TOOLCHAIN_ROOT "Zephyr toolchain root path invalid: please set the TOOLCH
 set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_TOOLCHAIN_VARIANT} CACHE STRING "Zephyr toolchain variant")
 assert(ZEPHYR_TOOLCHAIN_VARIANT "Zephyr toolchain variant invalid: please set the ZEPHYR_TOOLCHAIN_VARIANT-variable")
 
-if(CONFIG_ARCH_POSIX OR (ZEPHYR_TOOLCHAIN_VARIANT STREQUAL "host"))
+if(${ARCH} STREQUAL "posix" OR (ZEPHYR_TOOLCHAIN_VARIANT STREQUAL "host"))
```

Comment:
Separate PR?
If you're changing this anyway, why not clean it up so this special case is eliminated altogether?

Reply:
Will close this, do a separate PR, and then re-open.

Reply:
> If you're changing this anyway, why not clean it up so this special case is eliminated altogether?

This is out-of-scope of this PR. But a PR that did this would be welcome.